### ISSUE TITLE
Khoj fails to start because no Qt platform plugin could be initialized

### ISSUE THREAD

**Issue by digitalw00t** [khoj-ai/khoj#403]
[07:27:48] INFO     🌘 Starting Khoj                                                                                                                          main.py:66
qt.qpa.plugin: Could not load the Qt platform plugin "xcb" in "" even though it was found.
This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem.

Available platform plugins are: linuxfb, wayland-egl, wayland, minimalegl, xcb, eglfs, vnc, offscreen, vkkhrdisplay, minimal.

Aborted (core dumped)

This was installed with a venv using python -m pip install khoj-assistant.  This is running on a Linux Mint 21 machine (ubuntu 22).

**Comment by debanjum** [comment]
Are you able to start khoj without the `--gui` flag set?

**Comment by sabaimran** [comment]
@digitalw00t , are you on a Linux machine? Could you try running `sudo apt update && sudo apt install libegl1 libxcb-xinerama0 -y`? This installs some of the relevant gui dependencies.

**Comment by digitalw00t** [comment]
Yes I can run it without the --gui and it operates.  I executed sudo apt update && sudo apt install libegl1 libxcb-xinerama0 -y with no issues.  Same results from the --gui.  I'm not complaining, just trying to help with the debugging process.  And all packages were already installed, so there was no difference.

**Comment by debanjum** [comment]
I was able to finally reproduce this issue on a standard `Ubuntu 22.04 Server` when trying to install the Khoj debian package. To fix that error, I just set `export QT_QPA_PLATFORM=offscreen` and restarted the khoj application

@digitalw00t: 
1. Does setting `export QT_QPA_PLATFORM=offscreen` resolve this issue for you?
2. Were you trying to run it on a server (that is, a machine with no GUI or desktop environment setup)?
